srf_generation/source_parameter_generation/nshm2022_to_realisation.py

```diff
@@ -0,0 +1,421 @@
+#!/usr/bin/env python3
+"""
+Generate realisation stub files from ruptures in the NSHM 2022 database.
+
+This script generates YAML realisation stub files from ruptures in the NSHM 2022
+database. It extracts fault geometry computes causality information from the database
+and incorporates default parameter values to generate the realisation.
+
+Usage
+-----
+$ nshm2022_to_realisation.py NSHM_DB_FILE RUPTURE_ID YAML_FILE
+"""
+import collections
+import itertools
+import re
+from pathlib import Path
+from typing import Annotated, Any, TextIO, Tuple
+
+import numpy as np
+import qcore.coordinates
+import qcore.geo
+import qcore.uncertainties.mag_scaling
+import rupture_propogation
+import scipy as sp
+import typer
+import yaml
+from nshmdb import nshmdb
+from nshmdb.fault import FaultPlane
+from rupture_propogation import RuptureCausalityTree
+
+from srf_generation.realisation import RealisationFault
+
+
+def closest_points_between_faults(
```

Comment:
I think the "generic" functions in this script should sit in a module, since I suspect we will have a second version of this script for historic ruptures (since they won't be coming from the NSHM DB)